rust/agents/relayer/src/relayer.rs

```diff
@@ -64,8 +64,10 @@ impl BaseAgent for Relayer {
         let blacklist = parse_matching_list(&settings.blacklist);
         info!(whitelist = %whitelist, blacklist = %blacklist, "Whitelist configuration");
 
+        // TODO: we should support unknown chains
```

Comment:
This feels like a big regression that I'm really hesitant to merge with especially with our push for permissionless deployment (see https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1450). Iiuc previously someone could run a relayer or validator for a chain by only making config changes, and now it'd require code changes.

I think an easy way to keep the same functionality could be:
```
let origin_chain = core.settings.chain_setup(&settings.originchainname)?.domain()?;
```
And to make the `domain` fn pub. Same idea for the validator. Wdyt?

Alternatively could consider iterating through `core.mailboxes` or something and finding the `HyperlaneDomain` that way

Reply:
Yeah, I like this, done.